### ISSUE TITLE
support for curl meta-info on taskDefinition

### ISSUE THREAD

**Issue by cordoval** [aws/amazon-ecs-agent#274]
can we support curl meta-info resolving on taskDefinition?
i want to spawn a docker container and specify --dns to be $(curl http://.... metadata endpoint here from aws), i couldn't get it to work, how to point it to the host in which this docker container is running from a task definition dnsServers: ["??, ??"] 

thanks

**Comment by samuelkarp** [comment]
Hi @cordoval, can you provide more detail on your request?  What is your use-case?

**Comment by cordoval** [comment]
yeah, so for instance in a task definition like 

```
[
  {
    "name": "project",
    "image": "022.dkr.ecr.us-east-1.amazonaws.com/file-management",
    "cpu": 512,
    "memory": 512,
    "links": [],
    "portMappings": [
      {
        "containerPort": 8084,
        "hostPort": 8083,
        "protocol": "tcp"
      }
    ],
    "dnsServers": ["10.0.1.72", "8.8.8.8", "8.8.4.4"],
    "essential": true,
    "command": [],
    "environment": [
      { "name" : "LOG_LEVEL", "value" : "info" },
      // ...
    ],
    "entryPoint": [],
    "mountPoints": [],
    "volumesFrom": []
  }
]
```

in the line that goes like     "dnsServers": ["10.0.1.72", "8.8.8.8", "8.8.4.4"],

for instance one would like to use curl http://a.... metainfo/host-ip or something which is already supported.

However the on the template you cannot resolve a $(curl ...) on that line.

**Comment by samuelkarp** [comment]
Are you looking specifically to resolve the host IP or are you looking for arbitrary command execution?

**Comment by cordoval** [comment]
neither. Basically inside the task definition everything has to be resolved right now in order to work. I would like to use meta info resolving capabilities inside task definitions. Does it make sense?

**Comment by samuelkarp** [comment]
Right, so for the `dnsServers` field, I take it you want the host IP?

**Comment by cordoval** [comment]
yeah this is already supported in say cloudformation

```
$(curl -s http://169.254.169.254/latest/meta-data/local-ipv4)
```

and also if you run shell commands on a script

**Comment by samuelkarp** [comment]
Thanks.  It sounds like this is a very similar feature request to #3, except that you're looking for resolution in fields other than the environment variables.

**Comment by cordoval** [comment]
the solution would be different though as one of them is env vars and the other is resolving in syntax files you already handle. This is also handled again in cloudformation so i believe the parsing stuff is in aws side. :+1: thanks @samuelkarp

**Comment by willejs** [comment]
+1
This feature would be killer. I currently need to set the DNS server and hostsfile entry to the hosts IP inside the docker container. Im currently hacking around this and its really horrible.

**Comment by willejs** [comment]
@samuelkarp sorry to bother you, is this on the roadmap, or in discussion in your team? If so is there any sort of ETA, even a rough estimate?

**Comment by samuelkarp** [comment]
@willejs As a general rule, we don't comment on the future direction of the service, but we'd be interested in hearing the details of your use case.  Can you explain a bit more about what you would like to see specifically?

**Comment by cleung2010** [comment]
:+1:

**Comment by AjitDas** [comment]
@samuelkarp If I want to get the bunded hostPort mapped to a docker container port inside the docker container then I need something to inspect (ideal way would be the way we are able to get instance metadata information via http endpoint).

My use case is am using a hazelcast application in a docker container and I need to publish the <dockerhost/ec2_private_ip:hostport> combination when hazelcast starts so that it can publish the address to join in the hazelcast cluster. Well people recommend to use hardcoded hostPort instead of defining "0" but then it will limit ECS to scale more than one container inside a given instance because of hardcoded conflcit host port. It's a very critical and common for any docker apps these days even to introspect these infos, many others provide it inside docker via env variable, most of other docker orchestrator provides a way to get the info (for example in case of mesosphere we can get this info via env variable PORT_5701, PORT_8080 or PORT1, PORT2 etc). If AWS can provide something similar to that EC2 meadata or equivalent of accessing docker remote rest APIs also would be good enough.
